src/main/pages/base_page.py

```diff
@@ -82,17 +82,31 @@ def switch_to_iframe(self, by_locator):
     def return_from_iframe(self):
         self.driver.switch_to.default_content()
 
-    def switch_to_new_window(self):
+    def wait_for_element_to_disappear(self, by_locator, timeout=10):
+        WebDriverWait(self.driver, timeout).until(
+            expected_conditions.invisibility_of_element_located(by_locator)
+        )
+
+    def switch_to_new_window(self, by_locator):
         main_window = self.driver.current_window_handle
+
         all_windows = self.driver.window_handles
         for window in all_windows:
             if window != main_window:
                 self.driver.switch_to.window(window)
+                self.wait_for_element_to_disappear(by_locator)
                 break
 
-    def switch_to_tab(self, tab_index):
+    def switch_to_tab(self, tab_index, by_locator):
         tabs = self.driver.window_handles
+
         self.driver.switch_to.window(tabs[tab_index])
+        self.wait_for_element_to_disappear(by_locator)
+
+    def back_to_main_tab(self):
```

Comment:
You can create a function for switching to a tab with this code: 

```python
        tabs = self.driver.window_handles
        self.driver.switch_to.window(tabs[tab_index])
```

A function to switch_to_main_tab: `switch_to_tab(0)`.
Finally, you can create a third one for the switch and wait for an element to disappear that calls `switch_to_tab(x)` and wait.

Reply:
With the change mentioned above, this would not be necessary